src/nvim/search.c

```diff
@@ -710,23 +710,15 @@ int searchit(
                 /* for empty match: advance one char */
                 if (matchcol == matchpos.col
                     && ptr[matchcol] != NUL) {
-                  if (has_mbyte)
-                    matchcol +=
-                      (*mb_ptr2len)(ptr + matchcol);
-                  else
-                    ++matchcol;
+                  matchcol += (*mb_ptr2len)(ptr + matchcol);
```

Comment:
Sure. I was shortly wondering why the fp notation was in use but forgot to check that.